sw/device/lib/dif/dif_i2c.c

```diff
@@ -474,19 +483,30 @@ dif_result_t dif_i2c_write_byte_raw(const dif_i2c_t *i2c, uint8_t byte,
   }
 
   uint32_t fmt_byte = 0;
-  fmt_byte = bitfield_field32_write(fmt_byte, I2C_FDATA_FBYTE_FIELD, byte);
   fmt_byte = bitfield_bit32_write(fmt_byte, I2C_FDATA_START_BIT, flags.start);
   fmt_byte = bitfield_bit32_write(fmt_byte, I2C_FDATA_STOP_BIT, flags.stop);
   fmt_byte = bitfield_bit32_write(fmt_byte, I2C_FDATA_READB_BIT, flags.read);
   fmt_byte =
       bitfield_bit32_write(fmt_byte, I2C_FDATA_RCONT_BIT, flags.read_cont);
   fmt_byte = bitfield_bit32_write(fmt_byte, I2C_FDATA_NAKOK_BIT,
                                   flags.suppress_nak_irq);
-  mmio_region_write32(i2c->base_addr, I2C_FDATA_REG_OFFSET, fmt_byte);
+
+  for (size_t i = 0; i < size; ++i) {
+    uint32_t reg =
+        bitfield_field32_write(fmt_byte, I2C_FDATA_FBYTE_FIELD, bytes[i]);
+    mmio_region_write32(i2c->base_addr, I2C_FDATA_REG_OFFSET, reg);
+
+    spin_while_status_bit(i2c, I2C_STATUS_FMTFULL_BIT, /*set*/ true);
```

Comment:
Small detail: this will spin even when sending the last byte even though we are not pushing bytes after that. Would it make sense to move the `spin_while_status_bit` before writing to `I2C_FDATA_REG`?